tests/unit/test_mixins.py

```diff
@@ -376,3 +380,86 @@ def test_is_admin_grader(self, preferences, expected_is_admin_grader):
             patched_prefs.return_value = preferences
 
             self.assertEqual(self.block.is_admin_grader, expected_is_admin_grader)
+
+
+@ddt.ddt
+class TestDashboardRootXBlockMixin(TestCase, TestWithPatchesMixin):
+    class DashboardRootXBlockMixinGuineaPig(DashboardRootXBlockMixin):
+        def __init__(self):
+            self._project_details = mock.Mock()
+
+        @property
+        def project_details(self):
+            return self._project_details
+
+    def setUp(self):
+        self.block = self.DashboardRootXBlockMixinGuineaPig()
+        self.project_api_mock = mock.Mock(spec=TypedProjectAPI)
+        self.make_patch(
+            self.DashboardRootXBlockMixinGuineaPig, 'project_api',
+            mock.PropertyMock(return_value=self.project_api_mock)
+        )
+
+    @ddt.data(
+        [1, 2, 3],
+        [4, 5],
+        []
+    )
+    def test_workgroups(self, workgroup_ids):
+        def _get_workgroup_by_id(workgroup_id):
+            return {"id": workgroup_id, "users": []}
+
+        self.block.project_details.workgroups = workgroup_ids
+        self.project_api_mock.get_workgroup_by_id.side_effect = _get_workgroup_by_id
+
+        workgroups = self.block.workgroups
+
+        expected_calls = [mock.call(workgroup_id) for workgroup_id in workgroup_ids]
+        expected_groups = [_get_workgroup_by_id(workgroup_id) for workgroup_id in workgroup_ids]
+        self.assertEqual(self.project_api_mock.get_workgroup_by_id.mock_calls, expected_calls)
+        self.assertEqual(workgroups, expected_groups)
+
+    @ddt.data(
+        ([1], [1]),
+        ([2], [2, 3]),
+        ([1, 2], [1, 2, 3]),
+    )
+    @ddt.unpack
+    def test_users_in_workgroups(self, workgroup_ids, expected_user_ids):
+        workgroups = {
+            1: WorkgroupDetails(id=1, users=[{'id': 1}]),
+            2: WorkgroupDetails(id=2, users=[{'id': 2}, {'id': 3}])
+        }
+
+        def _get_workgroup_by_id(workgroup_id):
+            return workgroups.get(workgroup_id, None)
+
+        self.block.project_details.workgroups = workgroup_ids
+        self.project_api_mock.get_workgroup_by_id.side_effect = _get_workgroup_by_id
+
+        users = self.block.all_users_in_workgroups
+
+        self.assertEqual([user.id for user in users], expected_user_ids)
+
+    @ddt.data(
+        ({}, list('workgroups_sentinel'), list('user_sentinel')),
```

Comment:
@bradenmacdonald 

`['abc']` -> a list with single item 'abc'
`list('abc')` -> a list with three items 'a', 'b' and c.

This is a slight hack to fix the test after adding implicit `list` conversion in _append_context_parameters`.